### DISCUSSION TITLE
Enabling Mac Mouse Fix

### DISCUSSION THREAD

**Discussion by noah-nuebling** [noah-nuebling/mac-mouse-fix#861] [Guides]
If you ran into a problem with enabling Mac Mouse Fix, this guide will help you solve it.

> [!NOTE]
> **Don't read this guide if you're on macOS 15 Sequoia or above, or if you're on macOS 12 Monterey or below**
> The issues and solutions steps discussed in this guide only apply to macOS 13 Ventura and macOS 14 Sonoma. \
> If you're on another macOS version, please leave a comment below, so I can help you out and update this guide. Thanks!

## Why doesn't enabling Mac Mouse Fix work?

With macOS 13 Ventura, Apple introduced a new way for apps to run in the background called 'SMAppService'. Issues with this new system can cause problems with enabling Mac Mouse Fix. Especially if you have more than one copy of Mac Mouse Fix installed.

Update: Under macOS 15 Sequoia, Apple seems to have fixed these issues! If you're still experiencing issues with enabling Mac Mouse Fix, please leave a comment below so I can help you out and update this guide. Thank you.

**Additional Info**

Overall, I think the the new SMAppService system is a great improvement. It is much easier to use for developers and it gives users a way to see and control which apps run in the background at [System Settings > General > Login Items](https://noah-nuebling.github.io/redirection-service/?target=macos-settings-loginitems).

However, the new SMAppService system unfortunately has some bugs at the moment. Under certain circumstances, such as when you have two different copies of Mac Mouse Fix on your computer, the SMAppService system can get really confused. And then enabling Mac Mouse Fix won't work anymore.

Mac Mouse Fix 3 and later try to automatically detect these issues and provide solution steps right in the app. However, sometimes this is not possible. For those cases, or if you're using Mac Mouse Fix 2, this guide provides steps to solve the issue.

## The Solution

Below, you will find two solutions. You might have to use one or both of the solutions to solve the issue. 

The 1. Solution is simpler and should work in most circumstances. So I recommend starting with that. However, the 1. Solution will enable all "Allow in the Background" items under [System Settings > General > Login Items](https://noah-nuebling.github.io/redirection-service/?target=macos-settings-loginitems). If you'd like to avoid that, you might want to start with the 2. Solution.

### 1. Solution: Reset the Background-App Database

This solution resets a specific part of your computer's system that manages background apps. For most users this will fix the issue with Mac Mouse Fix.

1. Open the Terminal app. 
3. Enter the following command and press Return:
    ```
    sudo sfltool resetbtm
    ``` 
4. When prompted, enter your password.
5. Restart your computer.
6. Try enabling Mac Mouse Fix again.

**Additional Info**

- In case this approach doesn't work, try the **2. Solution**.
- The `sfltool resetbtm` command will enable all "Allow in the Background" items under [System Settings > General > Login Items](https://noah-nuebling.github.io/redirection-service/?target=macos-settings-loginitems). If you'd like to avoid that, you might want to try the **2. Solution** first.
- Apple recommends using the `sfltool resetbtm` command in this [support document](https://support.apple.com/guide/deployment/depdca572563/web), so you can be sure it is safe to use. 

### 2. Solution: Remove Duplicate Copies of Mac Mouse Fix

macOS might be trying to enable a different copy of Mac Mouse Fix than the one you intend to use. To fix this issue, you can delete all duplicate copies of Mac Mouse Fix and restart your computer:

1. Find duplicate copies of Mac Mouse Fix.
    - You can search for "Mac Mouse Fix" using Spotlight or use the Finder to find duplicate copies of Mac Mouse Fix.
    - To find the copy of Mac Mouse Fix that macOS is attempting to enable (and which might be causing these issues), enter the following command in the Terminal app and press Return:
    ```
    sudo sfltool dumpbtm | perl -ne 'print if /Mac%20Mouse%20Fix(?!%20Helper)[^\/]*\.app/' | sed 's/URL: //' | xargs open -R
    ```
    - This command should open a Finder window showing the copy of Mac Mouse Fix that macOS is trying to enable. If this command doesn't reveal anything in the Finder, you might have to restart your computer first for the command to work. 
3. Move any duplicate copies of Mac Mouse Fix which you found to the Trash.
4. Empty the Trash.
5. Restart your computer.
6. Try to enable Mac Mouse Fix again.

**Additional Info**

- In case this approach doesn't work, try the **1. Solution** instead.

---

I hope this helps! If you have any further questions or suggestions, let me know in a comment below.

*This guide was written with the help of ChatGPT.*

**Comment by Brtn11** [comment]
Hi love your app but mine stopped working (free version in turkey, thanks for the free version also its great im looking forward to tip or buy the app) I have Logi options installed when I quit Logi it starts working It was working earlier with Logi running in the background but now its not working( even the smooth scroll feature) but I have an another app like this named Mos and It works perfectly fine with Logi running in the background. Do you have a suggestion? Thanks .

**Reply by noah-nuebling** [reply]
Hey @Brtn11, thanks for you feedback, I'm glad you love the app! And I'm sorry for the late reply. 

Mac Mouse Fix isn't made to work alongside other mouse drivers such as Logitech Options. This is due to the way that Mac Mouse Fix captures and simulates mouse events. I plan to make this better at some point, but it'll be a lot of work, and I have so many things I want to do for the app first, so it's probably not coming soon.

In the mean time, I can recommend [Mos](https://mos.caldis.me/), [LinearMouse](https://linearmouse.app/) or [Steer Mouse](https://plentycom.jp/en/steermouse/). Those are the best alternatives to Mac Mouse Fix that I'm aware of. I'm not entirely sure whether LinearMouse and Steer Mouse can coexist with Logitech Options, but I think so.

If there's anything specific from Logitech Options that you'd like to see in MMF, please also let me know.

I hope that helps a little, and let me know if you have more questions. ((although it might take me a long time to answer, I'm sorry.))

For more information and discussion, see the GitHub Issue Category https://github.com/noah-nuebling/mac-mouse-fix/labels/Logitech%20Options%20Coexistence

**Reply by WilliamIPark** [reply]
I also had the problem with Logi Options. It might be worth adding to the guide above, as I didn't actually spot this comment until after I found a fix.

**Comment by Brtn11** [comment]
Thanks for your reply! I’ll try those out for sure! Also I have a feature
on my mind is it possible to make combinations with buttons? For example;
clicking and holding the upper side button and scrolling zooms the page.
And clicking and holding the lower side button while scrolling scrolls side
ways. Like I said I don’t know if it’s possible but I think it’s a cool
feature to have.
I Appreciate your hardwork
Best regards
Bertan

**Comment by Brtn11** [comment]
Hi have you found a solution? If so what is your solution ? Can you share
with us?

**Comment by nestorph** [comment]
I'm running Sequoia and not able to enable Mac Mouse Fix.  Button is just grayed out.

**Reply by noah-nuebling** [reply]
Hey @nestorph, thanks for the feedback, and I'm sorry for the late reply. 

Have you been able to activate the app since? If not and if you'd like to help investigate a solution or get a refund, please [shoot me an email](https://redirect.macmousefix.com/?target=mailto-noah).

**Comment by Even-Guo** [comment]
我使用的是罗技M585鼠标，目前只能识别鼠标中键，无法识别按键4、按键5。请问该如何解决

**Comment by FrankyPrograma** [comment]
Im trying to active mac mouse fix with the key that i recived on my gmail and say "There is no connection to the internet" but i have connexion im writing this comment with the same computer xd

**Comment by billbonney** [comment]
I just had the original issue that was posted here happen on macOS 26.5.1. I'm using Mac Mouse 3.0.3

I was stuck not being able to reenable it in the app.

This fix was simple add the app back into the login items, and bingo, it was now enabled and working.

Thanks 👍
